core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -101,6 +101,9 @@ private final void init(byte r[], int rOff, int rLen, byte cf[], int cfOff, int
     colVisibility = copyIfNeeded(cv, cvOff, cvLen, copy);
     timestamp = ts;
     deleted = del;
+    if (!sanityCheckKey()) {
+      throw new IllegalArgumentException("Invalid Key entry. Key exceeds " + Integer.MAX_VALUE + " bytes.");
```

Comment:
Would be nice to include the size.  Also why not have the sanityCheck method throw the exception?